src/authorization.ts

```diff
@@ -20,44 +20,64 @@ export interface BulkDataAuth {
     importResources?: string[];
 }
 
+/**
+ * Contains information from incoming http request. As well as
+ * additional context information such use case specific auth handlers or
+ * operational information specific to this request.
+ */
+export interface RequestContext {
+    method: string;
+    url: string;
+    hostname: string;
+    headers: KeyValueMap;
+    contextInfo: KeyValueMap;
+}
+
 export interface VerifyAccessTokenRequest {
     accessToken: string;
     operation: TypeOperation | SystemOperation;
     resourceType?: string;
+    requestContext?: RequestContext;
     id?: string;
     vid?: string;
     bulkDataAuth?: BulkDataAuth;
 }
 
 export interface AuthorizationBundleRequest {
     userIdentity: KeyValueMap;
+    requestContext?: RequestContext;
     requests: BatchReadWriteRequest[];
 }
 
 export interface AllowedResourceTypesForOperationRequest {
     userIdentity: KeyValueMap;
+    requestContext?: RequestContext;
     operation: TypeOperation | SystemOperation;
 }
 
 export interface AccessBulkDataJobRequest {
     userIdentity: KeyValueMap;
+    requestContext?: RequestContext;
     jobOwnerId: string;
 }
 
 export interface ReadResponseAuthorizedRequest {
     userIdentity: KeyValueMap;
+    requestContext?: RequestContext;
     operation: TypeOperation | SystemOperation;
     readResponse: any;
 }
 
 export interface WriteRequestAuthorizedRequest {
     userIdentity: KeyValueMap;
+    requestContext?: RequestContext;
     operation: TypeOperation;
     resourceBody: any;
 }
 
 export interface GetSearchFilterBasedOnIdentityRequest {
     userIdentity: KeyValueMap;
+    requestContext?: RequestContext;

```

Comment:
One thing I noticed is that the code calling `getSearchFilterBasedOnIdentity` for example:
https://github.com/awslabs/fhir-works-on-aws-routing/blob/mainline/src/router/handlers/resourceHandler.ts#L68
doesn't seem to have access to `express.Response.locals` where the `RequestContext` instance would be kept. 
So without a lot of changes in `fhir-works-on-aws-routing` to pass `requestContext` around or exposing `express.Response.locals` this approach isn't really working. 
The easiest way, with the least amount of changes is adding the `requestContext` to the `userIdentity` in either the implementation of [verifyAccessToken](https://github.com/awslabs/fhir-works-on-aws-interface/blob/mainline/src/authorization.ts#L74) or in the [authz handler](https://github.com/awslabs/fhir-works-on-aws-routing/blob/mainline/src/app.ts#L71)

Reply:
I know it is a bit more painful but I would prefer to make it explicit the requestContext is being passed in. It looks like we did something similar for the userIdentity: https://github.com/awslabs/fhir-works-on-aws-routing/blob/77d3f78e916df6208d603d03198a7a5391c5d8ff/src/router/routes/genericResourceRoute.ts#L141

Reply:
OK, good to know which path to pursue. I'll start working getting those changes into `fhir-works-on-aws-routing`.